### PR TITLE
today page

### LINKED ISSUE #2: Today page
When user is on 'Today', they should see a list of habits they have scheduled for that day and be able to mark them as complete. They should also be able to go back to previous days using an arrow button or calendar.

### PR DESCRIPTION
**Pull request by navdm**
Fetches a document containing habit completions for user specified date from firestore and creates a real-time listener on changes to that document with the onSnapshot method. If the document does not exist - its created. Otherwise the data contained in the document is transformed to React state and passed into HabitTodos.

### CODE REVIEW

File: src/pages/today/index.tsx [hunk 1]
```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
...
+	const documentId = useRef<string>(null);
 
 	const completedHabits = habitTodos.filter((habit) => habit.isComplete);
 	const incompleteHabits = habitTodos.filter((habit) => !habit.isComplete);
 
 	const isDateToday = isToday(date);
 	const isDateYesterday = isYesterday(date);
 
+	const handleCheckHabit =
+		(habitId: string) => (e: React.ChangeEvent<HTMLInputElement>) =>
```
Comment (joeypedicini92) [inline src/pages/today/index.tsx:70]: Minor, but instead of `React.ChangeEvent` you could just import ChangeEvent from React on line 38

File: src/pages/today/index.tsx [hunk 2]
```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
...
+				datesCollection(),
+				where('date', '>=', startOfDay(date)),
+				where('date', '<=', endOfDay(date)),
+				limit(1)
+			),
+			async ({ empty, docs }) => {
+				if (empty) {
+					const day = new Intl.DateTimeFormat('en-US', {
+						weekday: 'long',
+					}).format(date);
```
Comment (joeypedicini92) [inline src/pages/today/index.tsx:85]: it's a bit messy to put this logic in here in terms of readability, I'd extract it to a helper function and just do like `const day = currentDay()`

Comment (joeypedicini92) [reply src/pages/today/index.tsx:85]: and actually you're only using it on line 89, so you could just inline it there

File: src/pages/today/index.tsx [hunk 3]
```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
...
+					const querySnapshot = await getDocs(
+						query(
+							habitsCollection(),
+							where('repeatDays', 'array-contains', day)
```
Comment (joeypedicini92) [inline src/pages/today/index.tsx:89]: `where('repeatDays', 'array-contains', currentDay())

File: src/pages/today/index.tsx [hunk 4]
```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
...
-			getDateDoc(date),
-			(doc) => {
-				if (doc.exists()) {
-					const { date, ...habits } = doc.data();
-					setHabitTodos(
-						Object.entries(habits).map(([id, habit]) => ({
-							id,
-							...(habit as Omit<HabitTodo, 'id'>),
-						}))
+		return onSnapshot(
```
Comment (joeypedicini92) [inline src/pages/today/index.tsx:74]: This useEffect is pretty long and hard to follow. I'd definitely recommend refactoring it, or at the very least just write a summary comment of what is happening in here. 

A couple rules I try to follow: 
* I usually have a lint rule that warns me for functions that are longer than 12 lines. So this would be a red flag
* Functions should be easily understood to someone reading the code. If it takes more than a minute to parse through and understand what a function is doing that is a red flag. 
* If I can't refactor it for some reason, add a summary block comment that explains what is happening in plain text

File: src/pages/today/index.tsx [hunk 5]
```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
...
+			query(
+				datesCollection(),
+				where('date', '>=', startOfDay(date)),
+				where('date', '<=', endOfDay(date)),
```
Comment (joeypedicini92) [inline src/pages/today/index.tsx:78]: any reason you can't do `where('date', '==', date)`?

File: src/pages/today/index.tsx (hunk 4 above)
Comment (joeypedicini92) [reply src/pages/today/index.tsx:74]: A good place to start is looking at the if/else (lines 82 - 115)

maybe something like this:

```
      if (empty) {
        await createNewDateDocument(date);
      } else {
        await handleExistingDateDocument(docs[0], setHabitTodos);
      }
```

**Comment by joeypedicini92** [comment]
😎 Deploy Preview | https://deploy-preview-3--trackhabit.netlify.app/

I'm still learning what netlify can do, this is pretty awesome

**Comment by joeypedicini92** [comment]
If I add a task and start tracking from today, should it then show up on the today page? It is not currently but I was expecting that it would

File: src/pages/today/index.tsx (hunk 4 above)
Comment (navDhammu) [reply src/pages/today/index.tsx:69]: I created a separate function for creating a date document to improve readability.

File: src/pages/today/index.tsx (hunk 5 above)
Comment (navDhammu) [reply src/pages/today/index.tsx:73]: I tried that but it doesn't work. It seems like in firebase, the '==' doesn't work for dates.

**Comment by navDhammu** [comment]
> If I add a task and start tracking from today, should it then show up on the today page? It is not currently but I was expecting that it would

If you create a new habit after already having visited the today page for today's date, that means the new habit won't show up on today page because the document for that page would have already been created. So I need to fix that functionality.

**Comment by joeypedicini92** [comment]
looked over your last 2 commits good changes 👍
